src/core/core.layoutService.js

```diff
@@ -10,24 +10,58 @@ module.exports = function(Chart) {
 	Chart.layoutService = {
 		defaults: {},
 
-		// Register a box to a chartInstance. A box is simply a reference to an object that requires layout. eg. Scales, Legend, Plugins.
+		/**
+		 * Register a box to a chartInstance.
+		 * A box is simply a reference to an object that requires layout. eg. Scales, Legend, Title.
+		 * @method Chart.layoutService.addBox
```

Comment:
Indeed, that works well for nested arguments, but I think here it's really an interface and it's quite weird to have these `left, top, right, bottom` properties part of the function inputs. This method expects an object that implements the `IBox` (or `ILayoutBox` or `ILayoutItem`) interface, as we have for `IPlugin`, `IPlatform` or `IEvent`. Also more convenient if we need to refer to that interface from somewhere in the doc.

Actually, it doesn't require so much more doc if you define the interface as I did for [`IEvent`](https://github.com/chartjs/Chart.js/blob/312773ba7b7e2acf6cd1273c92017d3c197a992b/src/platforms/platform.js#L59).